Sources/StreamChatUI/UIConfig+ChannelList.swift

```diff
@@ -5,47 +5,66 @@
 import UIKit.UICollectionViewLayout
 
 public extension _UIConfig {
-    struct ChannelListUI {
+    struct ChannelList {
+        /// A button used for creating new channels.
+        public var newChannelButton: UIButton.Type = _ChatChannelCreateNewButton<ExtraData>.self
+
+        /// The logic to generate a name for the given channel.
+        public var channelNamer: ChatChannelNamer.Type = ChatChannelNamer.self
+
         /// The collection view of the Channel List.
-        public var channelCollectionView: ChatChannelListCollectionView.Type = ChatChannelListCollectionView.self
+        public var collectionView: ChatChannelListCollectionView.Type = ChatChannelListCollectionView.self
 
         /// The collection view layout of the Channel List.
-        public var channelCollectionLayout: UICollectionViewLayout.Type = ListCollectionViewLayout.self
+        public var collectionLayout: UICollectionViewLayout.Type = ListCollectionViewLayout.self
+
+        /// A `UICollectionViewCell` subclass that shows channel information.
+        public var collectionViewCell: _ChatChannelListCollectionViewCell<ExtraData>.Type =
+            _ChatChannelListCollectionViewCell<ExtraData>.self
 
         /// The Cell Separator View.
-        public var channelCellSeparatorReusableView: UICollectionReusableView.Type = _CellSeparatorReusableView<ExtraData>.self
+        public var cellSeparatorReusableView: UICollectionReusableView.Type = _CellSeparatorReusableView<ExtraData>.self
 
         /// The base view of the channel list item to support swipeable gestures.
-        public var channelListSwipeableItemView: _ChatChannelSwipeableListItemView<ExtraData>.Type =
+        public var swipeableItemView: _ChatChannelSwipeableListItemView<ExtraData>.Type =
             _ChatChannelSwipeableListItemView<ExtraData>.self
 
+        /// The subviews that compose the `swipeableItemView`.
+        public var swipeableItemSubviews = SwipeableItemSubviews()
+
         /// A `ChatChannelSwipeableListItemView` subclass view that shows channel information.
-        public var channelListItemView: _ChatChannelListItemView<ExtraData>.Type = _ChatChannelListItemView<ExtraData>.self
+        public var itemView: _ChatChannelListItemView<ExtraData>.Type = _ChatChannelListItemView<ExtraData>.self
 
-        /// A `UICollectionViewCell` subclass that shows channel information.
-        public var channelViewCell: _ChatChannelListCollectionViewCell<ExtraData>.Type =
-            _ChatChannelListCollectionViewCell<ExtraData>.self
+        /// The subviews that compose the `itemView`.
+        public var itemSubviews = ItemSubviews()
 
-        /// A button used for creating new channels.
-        public var newChannelButton: UIButton.Type = _ChatChannelCreateNewButton<ExtraData>.self
+        /// The subviews of that compose the `itemView`.
+        public struct ItemSubviews {
```

Comment:
I think the subviews is useful to better organize each view and to not overpopulate the config. This will be especially useful for the message list.